packages/preview-middleware-client/test/unit/flp/init.test.ts

```diff
@@ -13,6 +13,28 @@ import { fetchMock, sapMock } from 'mock/window';
 import type { InitRtaScript, RTAPlugin, StartAdaptation } from 'sap/ui/rta/api/startAdaptation';
 import type { Scenario } from '@sap-ux-private/control-property-editor-common';
 import VersionInfo from 'mock/sap/ui/VersionInfo';
+import { CommunicationService } from '../../../src/cpe/communication-service';
+
+jest.mock('../../../src/i18n', () => {
+    return {
+        ...jest.requireActual('../../../src/i18n'),
+        getTextBundle: async () => {
+            return {
+                hasText: jest.fn().mockReturnValueOnce(true),
+                getText: jest.fn().mockReturnValueOnce('The application was reloaded because of changes in a higher layer.')
+            }
+        }
+    }
+});
+
+Object.defineProperty(window, 'location', {
+    value: {
+        ...window.location,
+        reload: jest.fn()
+    },
+    writable: true
+});
```

Comment:
When you define a property like this, you need to reset it to the original value at the end of the test in order not to break the other chain of tests that uses the same code.

I would suggest to move this inside the new test case and also clear this at the end of the expect statement.

Example:

```ts
const reloadSpy = jest.fn();
const location = window.location;
Object.defineProperty(window, 'location', {
     value: {
         reload: reloadSpy
     }
);

...

expect(...)
Object.defineProperty(window, 'location', {
     value: location
});
```

Reply:
Fixed in [c2c0266](https://github.com/SAP/open-ux-tools/pull/2536/commits/c2c0266b492ca697c56288d62a0cd94ec0933034)

Reply:
This issue doesn't seem to be resolved yet. It probably shouldn't be set in global scope, but just before/within tests where it is actually needed. Also cleanup should be in the `after*` blocks, because the test might fail and the code at the end of the test it self might never run and leave the environment in an inconsistent state.